wagtailio/core/blocks.py

```diff
@@ -179,3 +181,196 @@ class HomePageBlock(blocks.StreamBlock):
 
     class Meta:
         template = "core/blocks/home_page_block.html"
+
+
+class CTABlock(blocks.StructBlock):
+    cta_text = blocks.CharBlock(label="CTA text", max_length=255)
+    cta_page = blocks.PageChooserBlock(label="CTA page", required=False)
+    cta_url = blocks.URLBlock(label="CTA URL", required=False)
+
+    def clean(self, value):
+        struct_value = super(PageOrExternalLinkBlock, self).clean(value)
+
+        if not value.get("cta_page") and not value.get("cta_url"):
+            raise ValidationError(
```

Comment:
this needs to be `StructBlockValidationError` otherwise no error gets raised. See https://github.com/wagtail/wagtail/issues/7202

additionally, you need to pass on a dict with `ErrorList` for values

Reply:
Thanks, updated in af8ddfc6d328e1b25906e7672977574cf591e782 and f3751fde600de09043686218a674a322b0039046